packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { ulStyleOverrides, LEFT_RIGHT_OFFSET } from './styles';
+
+const sideNavLabelStyle = css`
+  display: block;
+  color: ${uiColors.green.base};
+  font-size: 12px;
+  font-weight: bold;
+  text-transform: uppercase;
+  padding: 5px ${LEFT_RIGHT_OFFSET}px 5px ${LEFT_RIGHT_OFFSET}px;
+`;
+
+interface SideNavGroupProps {
+  /**
+   * Content that will appear inside of SideNavGroup component.
+   */
+  children: React.ReactNode;
+  /**
+   * ClassName that will be applied to root SideNavGroup element.
+   */
+  className?: string;
+  /**
+   * Header text displayed above the SideNavGroup contents.
+   */
+  headerText?: string;
+  /**
+   * ClassName that will be applied to SideNavGroup's header element.
+   */
+  headerClassName?: string;
+}
+
+/**
+ * # SideNavGroup
+ *
+ * ```
+<SideNavGroup headerText="Section Header">
+  <SideNavItem>
+    Link Content
+  </SideNavItem>
+</SideNavGroup>
+ * ```
+ * @param props.children Content that will appear inside of SideNavGroup component.
+ * @param props.className ClassName that will be applied to root SideNavGroup element.
+ * @param props.headerText Header text displayed above the SideNavGroup contents.
+ * @param props.headerClassName ClassName that will be applied to SideNavGroup's header element.
+ *
+ */
+function SideNavGroup({
```

Comment:
Just want to get @brookescarlett to weigh in here – from an accessibility standpoint, this seems correct overall to me, but I could be wrong.